sdk/nodejs/serverless/function.ts

```diff
@@ -254,55 +270,51 @@ interface Package {
             }
+
+            const allPackages = new Set<string>(includedPackages);
+            for (const depName of Object.keys(root.package.dependencies)) {
+                allPackages.add(depName);
+            }
+
             const s = new Set<string>();
-            for (var pkg of packages) {
-                if (pkg[0] == '.') {
-                    // Relative path, we can include the file itself, but we cannot deterministically find the
-                    // transitive dependencies, so the user may need to explicitly include those.  We use
-                    // `require.resolve` to get the resolved path to the file/folder in case a reference like `./foo` is
-                    // used to refer to `./foo.js`.
-                    try {
-                        const resolvedPath = resolveSync(pkg, { basedir: path });
-                        const relativePath = filepath.relative(path, resolvedPath);
-                        s.add(relativePath);
-                    } catch (err) {
-                        console.warn(
-                            `Could not find module for relative path '${pkg}' in '${filepath.resolve(root.path)}'.`);
-                    }
-                } else if (pkg[0] == '/') {
-                    // Absolute path, this won't work, so warn and move on.
-                    console.warn(
-                        `Could not include module for absolute path '${pkg}' in '${filepath.resolve(root.path)}'.`);
-                } else {
-                    // Neither relative nor aboslute path, so expected to be a name that resovles to `node_modules` (or
-                    // to a builtin, but those were removed).  We can add the package and all its transitive
-                    // dependencies.
-                    addPackageAndDependenciesToSet(s, root, pkg);
-                }
+            for (const pkg of allPackages) {
+                addPackageAndDependenciesToSet(s, root, pkg);
             }
+
             resolve(s);
         });
     });
-}
 
-// addPackageAndDependenciesToSet adds all required dependencies for the requested pkg name from the given root package
-// into the set.  It will recurse into all dependencies of the package.
-function addPackageAndDependenciesToSet(s: Set<string>, root: Package, pkg: string) {
-    var child = findDependency(root, pkg);
-    if (!child) {
-        console.warn(`Could not include required dependency '${pkg}' in '${filepath.resolve(root.path)}'.`)
-        return;
-    }
-    s.add(child.path);
-    if (child.package.dependencies) {
-        for (let dep of Object.keys(child.package.dependencies)) {
-            addPackageAndDependenciesToSet(s, child, dep);
+    // addPackageAndDependenciesToSet adds all required dependencies for the requested pkg name from the given root package
+    // into the set.  It will recurse into all dependencies of the package.
+    function addPackageAndDependenciesToSet(s: Set<string>, root: Package, pkg: string) {
```

Comment:
Nit - I'll admit I don't personally like these nested functions - it makes me nervous that these is some shared closure-captured state that I'm not aware of, and makes the contract between functions less "obvious".

Reply:
Oh - and indeed - this does now subtly have a closure captured dependency on `excludedPackages` even though it's similar dependency on `s` could have been closure captured but was not.  I would prefer to just force these to be explicit dependencies by unnesting the functions.

Reply:
sure.  the capture of excludedPackages was exactly why i did this.  but i'm happy to just pass it along.